src/relay/connection.ts

```diff
@@ -159,9 +159,35 @@ class RelayConnection extends EventEmitter implements MultiaddrConnection {
 
     this.source = this.createSource()
 
-    this._stream.sink(this.sinkFunction())
+    let sinkCreator: Promise<void>
```

Comment:
Please add documentation to this block of logic.

Reply:
done